utils/docker/Dockerfile.leap.15

```diff
@@ -6,7 +6,7 @@
 #
 
 # Pull base image
-ARG BASE_DISTRO=opensuse/leap:15.3
+ARG BASE_DISTRO=opensuse/leap:15.4
```

Comment:
Do you have a reference?  This is the dockerfile so won't affect our Leap RPM builds and it won't affect GitHub actions docker builds as they're on 15.4 already, all it affects is users calling docker directly (and it moves them onto the same platform as our GHA builds) and the icc build we have in Jenkins.  Happy to remove from this PR is preferred though, if we don't make the change here we should include it in https://github.com/daos-stack/daos/pull/9439

Reply:
Something I think I heard from @brianjmurrell last week but it might be out of date or unrelated to this change and I'm sure you have your finger on the pulse better than I wrt CI and OS versions.